pkg/storage/legacysql/dualwrite/dualwriter.go

```diff
@@ -105,12 +105,13 @@ func (d *dualWriter) Create(ctx context.Context, in runtime.Object, createValida
 		if err != nil {
 			log.Error("unable to cleanup object in legacy storage", "err", err)
 		}
+		return storageObj, errObjectSt
```

Comment:
Let's just return the error, I don't think we should return the objects here.
```suggestion
		return nil, errObjectSt
```

Reply:
I think the pattern in this file is confusing :-D I fixed it in the original one but we used it again here.